modules/apps/organizations/organizations-test/src/testIntegration/java/com/liferay/organizations/object/system/test/OrganizationSystemObjectDefinitionManagerTest.java

```diff
@@ -0,0 +1,299 @@
+
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Pedro Leite
+ */
+@RunWith(Arquillian.class)
+public class OrganizationSystemObjectDefinitionManagerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_organizationSystemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager("Organization");
+	}
+
+	@Test
+	public void testAddBaseModel() throws Exception {
+		_assertCount(0);
+
+		long organizationId1 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		_assertCount(1);
+
+		long organizationId2 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		_assertCount(2);
+
+		long organizationId3 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		_assertCount(3);
+
+		long organizationId4 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		_assertCount(4);
+
+		long organizationId5 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		_assertCount(5);
+
+		_organizationLocalService.deleteOrganization(organizationId1);
+		_organizationLocalService.deleteOrganization(organizationId2);
+		_organizationLocalService.deleteOrganization(organizationId3);
+		_organizationLocalService.deleteOrganization(organizationId4);
+		_organizationLocalService.deleteOrganization(organizationId5);
+	}
+
+	@Test
+	public void testDeleteBaseModel() throws Exception {
+		long organizationId1 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		long organizationId2 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		long organizationId3 = _addBaseModel(
+			HashMapBuilder.<String, Object>put(
+				"comments", RandomTestUtil.randomString()
+			).put(
+				"name", RandomTestUtil.randomString()
+			).build());
+
+		_assertCount(3);
```

Comment:
With these additions on a `setUp` method you can start by asserting if there are three organization entries, and then you will keep with the same code bellow.

Reply:
I think that to do it this way would make the code more complex.